healthz/README.md

```diff
@@ -7,41 +14,239 @@ in distributed systems.
 The ability to also include "implementation specific" details about the health
 of the system or component is very standard for network devices in the form for
 "show tech", "diag" or "debug" commands. These are very useful for getting
-useful diagnostics about the system state to the implementers. Healthz exposes
+diagnostics about the system state to the implementers. Healthz exposes
 these interfaces as queriable endpoints to allow operators to validate "health"
-on components and if "unhealthy" gather implementor specific data to help triage
-or reproduce issues.
-
-## Design
-
-* The service definition provides a Get method which is correlated with a
-  /component in the OC tree. The Get method is responsible for getting the
-  current status of the requested component path. This will cause the system
-  to fetch the current state of component this may entail running specific
-  validations and gathering any relevant status of the component to reflect
-  the "health check"
-
-* The internal design details of the "health check" and the frequency of the
-  health check are up to the implementor to determine. This presentation of
-  the status in OC is proposed to be modeled as
-
-```yang
-component/healthz/status = Enum
-component/healthz/last-unhealthy = Timestamp
-component/healthz/unhealthy_count = Int64
+on components and if "unhealthy" gather implementor specific data to help
+triage or reproduce issues.
+
+## Design Background
+
+`Healthz` provides a means by which a user may initiate checks of health
+(through the `Check` RPC) or a system may report the results of a check that it
+has initiated of its own accord. 
+
+Following a health check occuring, a caller uses the `List` RPC to discover the
+artifacts that are assoicated with a component and its corresponding
+subcomponents. Each artifact reflects an entity that is required to debug or
+further root cause the fault that occurs with it.
+
+The `Artifact` RPC is used to retrieve specific artifacts that are listed by
+the target system in the `List` RPC. Once retrieved each artifact can be
+'acknowledged' by the client of the RPC. Acknowledged artifacts are no longer
+returned in a list of the corresponding artifacts, and a device may use this
+status as a hint to allow garbage collection of artifacts that are no longer
+relevant. The device itself is responsible for garbage collection any may,
+if necessary, garbage collect artifacts that are not yet acknowledged.
```

Comment:
they should not be cleared just because the gnoi server restarted

Reply:
Added a clarification -- I think generally these need to persist across component restarts too, since healthz information that corresponds to something like a rebooting linecard should not be removed as the LC reboots IMHO.

Reply:
so essentially we have to persist the events until they are garbage collected/acked.